tap_twitter_ads/streams.py

```diff
@@ -98,10 +98,17 @@ def get_bookmark(self, state, stream, default):
         )
         
     # to read bookmarks in sync mode     
-    def write_bookmark(self, state, stream, value):
+    def write_bookmark(self, state, stream, value, sub_type=None):
         if 'bookmarks' not in state:
             state['bookmarks'] = {}
-        state['bookmarks'][stream] = value
+            
+        if sub_type:
+            # Store bookmark value for each sub_type of tweets stream
+            state['bookmarks'][stream] = state['bookmarks'].get(stream, {})
```

Comment:
Explain empty dict default assignment in the comment.

Reply:
Added comments.